### PR TITLE
fix to pass string to tokenizer

### PR DESCRIPTION
**Pull request by Soeb-aryn**
The tokenizer was receiving a Dict instead of text, causing the code to break.

### CODE REVIEW

File: lib/sycamore/sycamore/transforms/merge_elements.py [hunk 1]
```diff
@@ -159,8 +159,8 @@ def __init__(self, tokenizer: Tokenizer, max_tokens: int, merge_across_pages: bo
         self.merge_across_pages = merge_across_pages
 
     def preprocess_element(self, element: Element) -> Element:
-        if element.type == "Image" and "summary" in element.properties:
-            element.data["token_count"] = len(self.tokenizer.tokenize(element.properties["summary"] or ""))
+        if element.type == "Image" and "summary" in element.properties and "summary" in element.properties["summary"]:
+            element.data["token_count"] = len(self.tokenizer.tokenize(element.properties["summary"]["summary"] or ""))
```
[the remark this answers is not in the document]
Comment (Soeb-aryn) [reply lib/sycamore/sycamore/transforms/merge_elements.py:163]: yes, I have added one more unit test.

File: lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py [hunk 2]
```diff
@@ -169,7 +169,6 @@ class TestGreedySectionMerger:
                 },
                 {
                     "type": "Image",
-                    "text_representation": "image1 on page 2 before text5",
```
Comment (dtecuci) [inline lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py:172]: I thought you were going to add "summary" : {"summary" : "image1 on page 2 before text5", ..}
Having only an image without a summary will test the new if branch, but there is no test to test the image with summary merging. Please add a second image with summary

File: lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py [hunk 3]
```diff
@@ -239,6 +238,52 @@ def test_merge_elements(self):
         assert e.text_representation == ("Section-header1 on page 3")
         assert e.properties == {"page_number": 3, "columns": None, "page_numbers": [3], "rows": None, "title": None}
 
+    def test_merge_elements_image_summarize(self):
```
Comment (dtecuci) [inline lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py:241]: how is this test different from previous?

File: lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py (hunk 2 above)
Comment (Soeb-aryn) [reply lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py:172]: added another test for with image summary which I forgot to add earlier.

File: lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py (hunk 3 above)
Comment (Soeb-aryn) [reply lib/sycamore/sycamore/tests/unit/transforms/test_merge_elements.py:302]: This is suppose to be with image summary, which I forgot to add while git committing.

File: lib/sycamore/sycamore/transforms/merge_elements.py [hunk 4]
```diff
@@ -256,30 +256,43 @@ def merge(self, elt1: Element, elt2: Element) -> Element:
...
+                        new_elt.text_representation = elt1.text_representation + "\n" + elt2.table.to_html()
+                        new_elt.data["token_count"] = tok1 + 1 + tok2
                 else:
                     new_elt.text_representation = elt1.text_representation
                     new_elt.data["token_count"] = tok1
             else:
                 # text + text
-                new_elt.text_representation = elt1.text_representation + "\n" + elt2.text_representation
+                new_elt_text_representation = "\n".join(
+                    filter(None, [elt1.text_representation, elt2.text_representation])
```
Comment (dtecuci) [inline lib/sycamore/sycamore/transforms/merge_elements.py:292]: what is the purpose of this? To filter out the empty text_representation for either elt1 or elt2?

**Review by dtecuci — APPROVED** [review]

File: lib/sycamore/sycamore/transforms/merge_elements.py (hunk 4 above)
Comment (Soeb-aryn) [reply lib/sycamore/sycamore/transforms/merge_elements.py:292]: yes, that's right

**Pull request merged**